system/jlib/jtrace.cpp

```diff
@@ -235,19 +235,19 @@ class CSpan : public CInterfaceOf<ISpan>
         if (ctxProps == nullptr)
             return false;
 
-        if (!isEmptyString(hpccGlobalId.get()))
-            ctxProps->setProp(kGlobalIdHttpHeaderName, hpccGlobalId.get());
+        const char * globalId = queryGlobalId();
+        ctxProps->setNonEmptyProp(kGlobalIdHttpHeaderName, globalId);
```

Comment:
Trivial, but I think you could now get rid of the variable globalId and write:
`ctxProps->setNonEmptyProp(kGlobalIdHttpHeaderName, queryGlobalId());`